### PR TITLE
Add a sub-repository for node-matter

### PR DESCRIPTION
**Pull request by mfucci**
A simple relative path seems to work fine.
matter.js should be built first

### CODE REVIEW

**Comment by Apollon77** [comment]
And how this is "npm published"? SOmehoe I miss that part

**Comment by Apollon77** [comment]
I think this approach is a "non standard way" and I imagine many issues with it.

I googled around today a lot and also found e.g this https://2ality.com/2021/07/simple-monorepos.html ... 

In my eyes we should have a way for now that makes it very easy for any user to install from GitHub, or we need to release way more often in the future.

I also think that "lerna" is maybe no longer the way to go (but an option to easiely manage). NPM/Yarn Workspaces should be the way to go I think ?! But yes I do not have any real experience with it ...

Some more maybe helpful links:

* https://github.com/Quramy/npm-ts-workspaces-example <-- I plan to test around with that tomorrow a bit , installing, publishing or such ... 
* https://daveiscoding.com/nodejs-typescript-monorepo-via-npm-workspaces
* https://blog.frankdejonge.nl/setting-up-a-typescript-mono-repo-for-scoped-packages/
* https://plainenglish.io/blog/monorepo-setup-with-npm-and-typescript

So when I see that right it all works - just versioning of the packages we need to take care manually (or use tools like lerna to automate that)

WDYT?

**Comment by mfucci** [comment]
> And how this is "npm published"? Somehow I miss that part because yes this works for local development but not really with npm publish ... Or ?!

I don't think there is any problem for npm publish:
```
cd node/
npm publish
```

"npm publish" will use the package.json in the current directory.

File: node/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+    "name": "matter-node.js",
```
[the remark this answers is not in the document]
Comment (mfucci) [reply node/package.json]: Yea, I have created the npm scope "@project-chip" for matter.js

Updated to a name that can be published

**Comment by mfucci** [comment]
I am seeing your point now:
"@project-chip/matter.js": "../" works for local development

But when we publish, this should be replaced with the actual version of "@project-chip/matter.js"
So this kind of force to always release @project-chip/matter.js and @project-chip/node-matter.js at the same time (which might be fine)

So yes, great if you can experiment with different approach and tell us what is the best one...
Let's keep it simple, if possible...

**Comment by Apollon77** [comment]
@mfucci Yeds exactly the .. reference do not work for publish.

I will check the workspace approach today hopefully and provide details.

I think a proces where someone that publishs needs to manually adjust stuff and revert that after publish is not that helpful in my eyes and error prone (and honestly might be problematic for automatic npm releases in the future ... In all my other projects we use automatic npm releases by github actions and by tagging a new version ;-) ... but discussing this is even another story) ;-)

**Comment by Apollon77** [comment]
Ooookkk ... I looked a bit into the "workspaces" idea and that seems to be the way to go to me. A good example -also for our case with Typescript is https://github.com/Quramy/npm-ts-workspaces-example ...

So as idea this would be like:

* anything right now in "matter.js" will be moved to "/packages/matter.js"
* anything from "node-matter" would go initially into "/packages/matter-node.js" (to stay with the idea from @mfucci)
* The root get the "Main meta package.json" (will never get published) that defines the workspace thing and contains global build files and such
* The package.json's of the workspace packages contain the package references as normal with their full names
* when changing dependencies in sub packages you NEVER execute "npm install" in the sub package dirs but only in the "main root directory" ... then npm will handle all stuff automatically and locally the packages are linked directly via relative paths
* npm run and also npm publish have since v7 support to also execute the commands then for all workspace packages, so with this you can publish all packages at once
* "lerna" can be used to do the version managament more convenient when publishing (if we like, else we need to manage versions manually)

lets merge the PRs here and I will provide an example PR for such a move with a minimal example like Marco did

**Comment by mfucci** [comment]
@Apollon77 : the complexity is that matter.js and matter-node.js compiles against a different set of APIs (pure ES6 vs node) and are using a different module system (ES6 vs commonJs).
Give it a shot, but the two problems above might make it tricky

I will keep this MR open for now just in case: should not be too hard to hack it to make publishing work (replacing "../" by latest matter.js version with a script before publishing).

**Comment by Apollon77** [comment]
@mfucci I hope we can somehow leave it that way "separated "... but need to see for this "global ts stuff" ... need to experiment. But yes leave open here for now ...

**Comment by Apollon77** [comment]
Because we decided for the approach in #52 I close here

**Pull request closed without merging**